### ISSUE TITLE
Cannot alter generated ddl

### ISSUE THREAD

**Issue by batousik** [jOOQ/jOOQ#7976]
### Expected behavior and actual behavior:

I ve expected to retrieve ddl from existing tables and then alter table and constraints names and then execute the statement to create a clone table, but fields are private/protected

### Steps to reproduce the problem (include example code if possible):

```java
// SCHEMA is the generated schema that contains a reference to all generated tables
Queries ddl =
DSL.using(configuration)
   .ddl(SCHEMA);
   
for (Query query : ddl.queries()) {
    System.out.println(query);
}

eg: query.getTable.setName(newName)
      quuery.execute();
```

### Versions:
- jOOQ: 3.11.5
- Java: 1.8
- Database (include vendor): Oracle 11.2.0.4.0
- OS: win 7
- JDBC Driver (include name if inofficial driver): com.oracle.ojdbc6:11.2.0.3

**Comment by lukaseder** [comment]
Thank you very much for your report. Indeed, this is currently not possible, and there are no plans of opening up those internals - for a variety of good reasons.

However, let's discuss your use-case, which sounds very interesting. What are you trying to achieve here?

**Comment by batousik** [comment]
I would like to create a clone of a table with a different name, constraint names and indexes names.

**Comment by lukaseder** [comment]
I see, thanks for the feedback. Indeed that `rename()` method could be expanded, in order to rename things recursively.

Alternatively, the runtime schema mapping also offers a partial solution to this, by allowing to rename schemas and tables using regular expressions. Perhaps, instead of inventing a new mechanism, we could improve on the schema mapping and allow for mapping all object types, including constraint and index names. This would allow for consistently using such a mapping for both:

- The creation of the clone
- Querying the clone from the original generated table

**Comment by lukaseder** [comment]
I will keep this issue open as a means to verify the correctness of the new feature request #7993, which will extend the `Settings` and its `RenderMapping` property

**Comment by batousik** [comment]
Looking forward to it!

**Comment by lukaseder** [comment]
Starting with jOOQ 3.16, the new (experimental) query object model (see https://github.com/jOOQ/jOOQ/issues/12425) will include methods to modify any `QueryPart`, recursively. The current API is `QueryPart.$replace()` (may change), and it could look like this:

```java
Queries ddl = ctx.ddl(S1);

for (Query query : ddl.queries()) {
    System.out.println(query);
    System.out.println(query.$replace(q -> {
        if (q instanceof Table<?> t)
            return table(name(t.getQualifiedName().qualifier(), name("copy_" + t.getName())));

        return q;
    }));
}
```

**Comment by lukaseder** [comment]
Closed as duplicate of #12425
